ckan/templates/user/request_reset.html

```diff
@@ -35,8 +35,9 @@ <h1 class="page-heading">{{ _('Reset your password') }}</h1>
     {% block help_inner %}
     <h2 class="module-heading">{{ _('How does this work?') }}</h2>
     <div class="module-content">
-      <p>{% trans %}Enter your username into the box and we will send you
-      an email with a link to enter a new password.{% endtrans %}</p>
+      <p>{% trans %}Enter your email address or username into the box and we
+        will send you an email with a link to enter a new password.
+        {% endtrans %}</p>
```

Comment:
Allowing users to enter their username is sensible for the case you've specified below (multiple users with the same email), but with the user listing being enabled by default, it essentially means an attacker has similar information available to them (complete listing of all the users, rather than complete listing of emails).

To patch this hole we'll need to get the users listing page disabled by default.

Reply:
Agreed. Let's not couple this PR to that piece of work though.